src/controller/CHIPDeviceController.cpp

```diff
@@ -2175,6 +2178,44 @@ void DeviceCommissioner::ParseFabrics()
     CommissioningStageComplete(return_err, report);
 }
 
+void DeviceCommissioner::ParseIcdInfo()
+{
+    CHIP_ERROR err;
+    IcdInfo info;
+    IcdManagement::Attributes::FeatureMap::TypeInfo::DecodableType featureMap;
+
+    // TODO(#29382): We probably want to read "ActiveMode" attribute (to be implemented) for ICD.
+    err = mAttributeCache->Get<IcdManagement::Attributes::FeatureMap::TypeInfo>(kRootEndpointId, featureMap);
+    if (err == CHIP_NO_ERROR)
+    {
+        info.isIcd                  = true;
+        info.checkInProtocolSupport = !!(featureMap & 0x1);
```

Comment:
Please use the actual enums generated for this, not hardcoded values.

Reply:
Updated